.gitattributes

```diff
@@ -0,0 +1 @@
+*.sh text eol=lf
```

Comment:
huh what is this? (I'm sure it's right, i'm just unfamiliar)

Reply:
On Windows, line endings are encoded as \r\n (which is typically written out as CRLF, or carriage return + line feed)

On Unix, line endings are encoded as \n (which is typically written out as LF, or line feed)

Git's strategy for dealing with this problem is to look at the host you're working on, and "fix" the line ending coding while you're editing, and then changing it back when you commit the code.

Finally, Unix shells aren't happy if you have CR characters in shell scripts.

These shell scripts are going to be copied into a unix container. We need a way to tell Git "don't convert *.sh files to CRLF encoding on Windows. We need them to have LF line endings"

This line is how we say that.